pkg/asset/agent/gencrypto/authconfig.go

```diff
@@ -54,12 +74,33 @@ func (a *AuthConfig) Generate(_ context.Context, dependencies asset.Parents) err
 
 	a.PublicKey = encodedPubKeyPEM
 
-	token, err := localJWTForKey(infraEnvID.ID, privateKey)
+	newAgentAuthToken, err := LocalJWTForKey(infraEnvID.ID, privateKey, agentWorkflow.Workflow)
 	if err != nil {
 		return err
 	}
-	a.Token = token
+	a.AgentAuthToken = newAgentAuthToken
+
+	switch agentWorkflow.Workflow {
+	case workflow.AgentWorkflowTypeInstall:
+		// Auth tokens do not expire
```

Comment:
This comment doesn't match the current activity, which is the creation (or refresh) of an auth token as a cluster secret. Please move this whole piece within a more speaking method.

Reply:
The comment is meant for the install workflow